### ISSUE TITLE
IntelliSense and analyzers node are *completely* broken with legacy style .csproj projects and manual Roslyn multi-targeting

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#66047]
**Version Used**: 17.5.0 Preview 3.0 [33213.424.main]

**Steps to Reproduce**:

1. Create a new legacy style .csproj project
2. Reference the MVVM Toolkit 8.1 Preview 2
3. Add the target from https://github.com/CommunityToolkit/dotnet/pull/541
4. Try to use some generated code, eg.
```csharp
public partial class Foo : ObservableObject
{
    [ObservableProperty]
    private int number;
}
```

**Expected Behavior**:

Things should just work as expected. The custom target is manually handling the component versioning.

**Actual Behavior**:

The project does build, but IntelliSense and the analyzers node are **completely** broken:
- Generated code isn't visible (doesn't show up in quick info)
- Referencing generated code is marked as an error, saying the reference members don't exist
- Analyzer warnings only show up in the output console but not in IntelliSense.
- Members referenced by generated code show up as unreferenced.
- The analyzers node in solution explorer is just completely empty and no analyzer is visible at all, so it's also impossible to inspect the generated code.

We can't really ship this in this state, as other than the fact it does build, it's effectively completely unusable. This is affecting several first party apps (eg. the Store, Photos, etc.) that are all using the MVVM Toolkit and the generators from UWP, which is stuck on legacy style .csproj projects. Plus, of course, all our other external customers of the MVVM Toolkit that also use legacy style projects.

@sharwell @chsienki @jasonmalinowski could anyone help with this? I'm at a loss on how to proceed with this 😅

I understand that Roslyn doesn't support component versioning on legacy style .csproj projects, and that's fine, our custom target is taking care of removing the analyzers already. But the issue is that everything's still half broken after that too 🥲

> **Note**: I tested this on VS 17.4.3 as well and things actually seem to work correctly there. Would it be possible to double check whether VS 17.5 Preview 3 acting up is just due to it being in preview and not actually a regression that it will introduce for customers using this setup once it ships? I would really like to avoid a situation where VS 17.5 is released, people update to that and then suddenly their IntelliSense becomes completely broken if using the MVVM Toolkit on legacy projects.

**Comment by Sergio0694** [comment]
Closing this for now as I can't seem to repro anymore with the latest preview and internal preview, when using the newly published 8.1.0 Preview 3 version of the MVVM Toolkit, which does include that custom .targets. Fingers crossed

**Comment by jasonmalinowski** [comment]
@Sergio0694: if you're able to reproduce again, do let us know!
